### ISSUE TITLE
Make Watchify optional

### ISSUE THREAD

**Issue by waldemarfm** [skrajewski/laravel-elixir-browserify#20]
There should be a way to make watchify optional for cases like mine (NFS mount) where it cannot be used. It should have the option to use the default elixir watcher as previous versions.

**Comment by Daveawb** [comment]
It is. Just don't use '.watchify()' and use the standard 'watch' task

**Comment by Daveawb** [comment]
In fact you can still include the `.watchify()` method but use `gulp watch` instead of `gulp watchify`

**Comment by waldemarfm** [comment]
@Daveawb I thought that would be thecase, but doing either of those makes the browserify task not even trigger at all, much less do any watching. Here's my gulp output after a  `gulp watch`:

```
[02:00:56] Using gulpfile ~/Code/nobox/website/gulpfile.js
[02:00:56] Starting 'watch'...
[02:00:56] Starting 'sass'...
[02:00:57] Finished 'watch' after 560 ms
[02:00:57] gulp-notify: [Laravel Elixir]
[02:00:57] Finished 'sass' after 1.23 s
[02:00:57] Starting 'version'...
[02:00:57] Finished 'version' after 79 ms
[02:00:58] Starting 'watch-assets'...
[02:00:58] Finished 'watch-assets' after 49 ms
```

Using `gulp watchify` on the other hand does run browserify, so that's why I'm stating that watchify is not optional right now that why it's implemented.

**Comment by Daveawb** [comment]
There is a bug there then, the regular watch shouldn't be interfered with.

**Comment by waldemarfm** [comment]
I've been playing around with the code on my own fork, and I think the issue is with the `config.watchify` used in several conditionals in the script, specifically the registerWatcher method.

`config` is an alias for `this` which is Elixir. Elixir stores all the tasks with the task name as the property name. An excerpt:

```
{ ...
browserify: [Function],
watchify: [Function],
svgSymbols: [Function],
imagemin: [Function],
... }
```
This means that `config.watchify` will always evaluate to true, regardless of having called the watchify task or not. When calling the default Elixir watch, the conditional in the third parameter of `registerWatcher` will always be true, registering the task to the `nowatch` group, thus disabling watching.

```
this.registerWatcher('browserify', options.srcDir + '/**/*.js', config.watchify ? 'nowatch' : 'default');
```

Using `gulp watch`, the same console log of `config` outputs: 


```
watchers:
 { default: { sass: 'resources/assets/sass/**/*.+(sass|scss)' },
   nowatch: { browserify: 'resources/assets/js/**/*.js' } },
```

In summary, relying on `config.watchify` to register the watcher or not is not the way to go. Maybe changing the flag to a standalone boolean variable instead of using config? Something like `useWatchify` that is `false` by default and changed to `true` if the watchify task is called?

**Comment by skrajewski** [comment]
@waldemarfm,

Thanks for your explanation, it was really helpful. This will be fixed in #22.
